modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/upgrade/v1_0_0/DeprecatedFeatureFlagUpgradeProcess.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.feature.flag.web.internal.upgrade.v1_0_0;
+
+import com.liferay.feature.flag.web.internal.company.feature.flags.CompanyFeatureFlags;
+import com.liferay.feature.flag.web.internal.company.feature.flags.CompanyFeatureFlagsProvider;
+import com.liferay.feature.flag.web.internal.model.FeatureFlag;
+import com.liferay.feature.flag.web.internal.model.FeatureFlagType;
+import com.liferay.portal.events.StartupHelperUtil;
+import com.liferay.portal.kernel.model.CompanyConstants;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.ReleaseInfo;
+
+import java.util.List;
+
+/**
+ * @author Thiago Buarque
+ */
+public class DeprecatedFeatureFlagUpgradeProcess extends UpgradeProcess {
+
+	public DeprecatedFeatureFlagUpgradeProcess(
+		CompanyFeatureFlagsProvider companyFeatureFlagsProvider,
+		CompanyLocalService companyLocalService) {
+
+		_companyFeatureFlagsProvider = companyFeatureFlagsProvider;
+		_companyLocalService = companyLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (_disableDeprecatedFeatures()) {
+			CompanyFeatureFlags systemFeatureFlags =
+				_companyFeatureFlagsProvider.getOrCreateCompanyFeatureFlags(
+					CompanyConstants.SYSTEM);
+
+			List<FeatureFlag> deprecationFeatureFlags =
+				systemFeatureFlags.getFeatureFlags(
+					FeatureFlagType.DEPRECATION.getPredicate());
+
+			_companyLocalService.forEachCompanyId(
+				companyId -> {
+					for (FeatureFlag deprecationFeatureFlag :
+							deprecationFeatureFlags) {
+
+						_companyFeatureFlagsProvider.setEnabled(
+							companyId, deprecationFeatureFlag.getKey(), false);
+					}
+				});
+		}
+	}
+
+	private boolean _disableDeprecatedFeatures() {
+		if (ReleaseInfo.isDXP() && !StartupHelperUtil.isDBNew() &&
+			StartupHelperUtil.isUpgrading()) {
```

Comment:
I discussed, this one with Mateo, and we don't need to include the `ReleaseInfo.isDXP()` check. We want the same behavior for Portal and DXP.

Reply:
Also this check might be clearer if we invert it:
```
if (StartupHelperUtil.isDBNew() && !StartupHelperUtil.isUpgrading()) {
     return true;
}
else {
    return false;
}